File: Card.java
public class Card {

    public enum Suit {
        CLUBS, DIAMONDS, HEARTS, SPADES;
    }

    public static final int JACK = 11;
    public static final int QUEEN = 12;
    public static final int KING = 13;
    public static final int ACE = 14;

    private Suit suit;
    private int rank;

    public Card(Suit suit, int rank) {
        this.suit = suit;
        this.rank = rank;
    }

    @Override
    public String toString() {
        String ret = "";
        switch (rank) {
            case JACK:
                ret += "JACK";
                break;
            case QUEEN:
                ret += "QUEEN";
                break;
            case KING:
                ret += "KING";
                break;
            case ACE:
                ret += "ACE";
                break;
            default:
                ret += Integer.toString(rank);
        }
        return ret + "of" + suit.name();
    }

    public Suit getSuit() {
        return suit;
    }

    public int getRank() {
        return rank;
    }

    public static Card fromString(String str) {
        String rs = str.substring(0, str.indexOf('o'));
        int r;
        if (rs.equals("JACK")) {
            r = 11;
        } else if (rs.equals("QUEEN")) {
            r = 12;
        } else if (rs.equals("KING")) {
            r = 13;
        } else if (rs.equals("ACE")) {
            r = 14;
        } else {
            r = Integer.parseInt(rs);
        }
        return new Card(Suit.valueOf(str.substring(str.indexOf('f') + 1)), r);
    }
}

File: src/main/java/com/isaac/games/Card.kt
package com.isaac.games

open class Card(val suit: Suit, val rank: Int) {

    enum class Suit {
        CLUBS, DIAMONDS, HEARTS, SPADES
    }

    init {
        if (rank < 2 || rank > 14) {
            throw IllegalArgumentException("Bad rank: $rank")
        }
    }

    override fun equals(o: Any?): Boolean {
        if (o !is Card) {
            return false
        }
        val that = o as Card?
        return this.suit == that!!.suit && this.rank == that.rank
    }

    override fun toString(): String {
        var ret = ""
        when (rank) {
            JACK -> ret += "JACK"
            QUEEN -> ret += "QUEEN"
            KING -> ret += "KING"
            ACE -> ret += "ACE"
            else -> ret += Integer.toString(rank)
        }
        return ret + "of" + suit.name
    }

    companion object {

        val JACK = 11
        val QUEEN = 12
        val KING = 13
        val ACE = 14

        fun fromString(str: String): Card {
            val rs = str.substring(0, str.indexOf('o'))
            val r: Int
            if (rs == "JACK") {
                r = 11
            } else if (rs == "QUEEN") {
                r = 12
            } else if (rs == "KING") {
                r = 13
            } else if (rs == "ACE") {
                r = 14
            } else {
                r = Integer.parseInt(rs)
            }
            return Card(Suit.valueOf(str.substring(str.indexOf('f') + 1)), r)
        }
    }
}